src/pages/en/guides/imports.md

```diff
@@ -73,17 +77,20 @@ While Astro understands JSX syntax out-of-the-box, you will need to include a fr
 **Astro does not support JSX in `.js`/`.ts` files.** JSX will only be handled inside of files that end with the `.jsx` and `.tsx` file extensions.
 :::
 
-## NPM Packages
+### NPM Packages
 
-```js
-// Returns the React & React-DOM npm packages
-import React from 'react';
-import ReactDOM from 'react-dom';
+```astro
+---
+import { Icon } from 'astro-icon';
+---
 ```
+If you've installed an NPM package, you can import it in Astro. Even if a package was published using a legacy format, Astro will convert the package to ESM so that `import` statements work.
 
-Astro lets you import npm packages directly in the browser. Even if a package was published using a legacy format, Astro will up-convert the package to ESM before serving it to the browser.
+:::warning
+Some packages rely on a browser environment. Astro components runs on the server, so importing these packages in the frontmatter may lead to errors.
```

Comment:
Love this! Just checking, do they "lead to errors" or do they just.... "silently not work"? Not do what you expect them to.

Reply:
Perfect, made an issue for a troubleshooting guide entry!